### PR TITLE
@unbox Pandas.Series -> StringArray

### PR DESCRIPTION
**Pull request by fschlimb**
@ehsantn

### CODE REVIEW

File: hpat/_str_ext.cpp [hunk 1]
```diff
@@ -311,6 +316,68 @@ void* str_from_float64(double in)
...
+
+    uint32_t * offsets = new uint32_t[n];
+    std::vector<const char *> tmp_store(n);
+    size_t len = 0;
+    for(Py_ssize_t i = 0; i < n; ++i) {
+        offsets[i] = len;
+        PyObject * s = PySequence_GetItem(obj, i);
+        CHECK(s, "getting element failed");
+        CHECK(PyString_Check(s), "expecting a string");
+        tmp_store[i] = PyString_AsString(s);
```
Comment (ehsantn) [inline hpat/_str_ext.cpp]: @fschlimb I think we need to deallocate the strings returned from `PyUnicode_AsUTF8`

https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_AsUTF8AndSize

**Comment by ehsantn** [comment]
@fschlimb could you add a unittest for this?

File: hpat/_str_ext.cpp (hunk 1 above)
Comment (fschlimb) [reply hpat/_str_ext.cpp]: the doc says: "The caller is not responsible for deallocating the buffer."
We release the reference to the PyObject in 363.

Comment (ehsantn) [reply hpat/_str_ext.cpp]: Sorry, I misread the page :)

**Comment by fschlimb** [comment]
a meaningful unittest is not that straight-forward as we need a PyObject and it returns a C-StringArray.

I can provide a small python example which tests it, though.

**Comment by ehsantn** [comment]
Let's add unittests with the next PR. merging.

**Pull request merged**